deflate.c

```diff
@@ -593,7 +593,10 @@ int32_t ZEXPORT PREFIX(deflatePrime)(PREFIX3(stream) *strm, int32_t bits, int32_
         put = BIT_BUF_SIZE - s->bi_valid;
         if (put > bits)
             put = bits;
-        s->bi_buf |= (((uint64_t)value & ((UINT64_C(1) << put) - 1)) << s->bi_valid);
+        if (s->bi_valid == 0)
+            s->bi_buf = (uint64_t)value;
```

Comment:
Ignoring ```bits``` is safe here because below we essentially set ```s->bi_valid```  to value of ```put``` (actually ```0 + put```) and the test ```put > bits``` above is outside the ```if``` block... Any extra bits will be shifted away when adding more bits to ```s->bi_buf```.

Reply:
@mtl1979, was there something you wanted me to do or were you just commenting?

Reply:
@nmoinvaz I was just wondering if it needs a comment to explain why it ignores ```bits``` when ```s->bi_valid``` is 0.

Reply:
I don't really think so personally.